src/services/expensesService.js

```diff
@@ -0,0 +1,54 @@
+let expenses = [];
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+const getAllExpenses = (userId, categories, from, to) => {
+  return expenses
+    .filter((expense) => !userId || expense.userId === +userId)
+    .filter((expense) => !categories || expense.category === categories)
+    .filter((expense) => !from || new Date(expense.spentAt) >= new Date(from))
+    .filter((expense) => !to || new Date(expense.spentAt) <= new Date(to));
+};
+
+const getExpenseById = (id) => {
+  return expenses.find((expense) => expense.id === +id);
+};
+
+const createExpense = (userId, spentAt, title, amount, category, note) => {
+  const expense = {
+    id: Math.floor(Math.random() * 1000),
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  expenses.push(expense);
+
+  return expense;
+};
+
+const updateExpense = (id, data) => {
+  const expense = getExpenseById(id);
+
+  Object.assign(expense, data);
+
+  return expense;
```

Comment:
Here you are directly modifying the expense object found by the `getExpenseById` function. This can lead to unexpected behavior if the expense object is used elsewhere in your code. Consider creating a new object with the updated values instead of modifying the existing one.